crypto/verifier.go

```diff
@@ -31,8 +31,13 @@ import (
 	"github.com/google/trillian/crypto/sigpb"
 )
 
-// ErrVerify occurs whenever signature verification fails.
-var ErrVerify = errors.New("signature verification failed")
+var (
+	errVerify = errors.New("signature verification failed")
```

Comment:
In other languages, identifying a type of error based on the error message would be frowned upon. Is this another of those cases where Go goes against convention? In most languages, you would basically use type assertions to identify a particular kind of error (and therefore have something like `type VerifyError struct { Error }`)?